tracer/src/Datadog.Trace/Telemetry/TelemetryControllerV2.cs

```diff
@@ -211,17 +210,12 @@ private async Task PushTelemetryLoopAsync()
 
             if (_sendTelemetry)
             {
-                await PushTelemetry().ConfigureAwait(false);
+                await PushTelemetry(sendAppClosing: _processExit.Task.IsCompleted).ConfigureAwait(false);
```

Comment:
Although minimal, isn't there a risk that IsCompleted is set to true before the if below?

Reply:
Yeah, good point, I should extract that to a variable, thanks!